spec/integration/browser_context/base_url_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'spec_helper'
+
+RSpec.describe 'baseUrl' do
+  it 'should construct a new URL when a baseURL in browser.newContext is passed to page.goto', sinatra: true do
+    with_context(baseURL: server_prefix) do |context|
+      page = context.new_page
+      expect(page.goto('/empty.html').url).to eq(server_empty_page)
+    end
+  end
+
+  it 'should construct a new URL when a baseURL in browser.newPage is passed to page.goto', sinatra: true do
+    with_page(baseURL: server_prefix) do |page|
+      expect(page.goto('/empty.html').url).to eq(server_empty_page)
+    end
+  end
+
+  it 'should construct the URLs correctly when a baseURL without a trailing slash in browser.newPage is passed to page.goto', sinatra: true do
+    with_page(baseURL: "#{server_prefix}/url-construction") do |page|
+      expect(page.goto('mypage.html').url).to eq("#{server_prefix}/mypage.html")
+      expect(page.goto('./mypage.html').url).to eq("#{server_prefix}/mypage.html")
+      expect(page.goto('/mypage.html').url).to eq("#{server_prefix}/mypage.html")
+    end
+  end
+
+  it 'should construct the URLs correctly when a baseURL with a trailing slash in browser.newPage is passed to page.goto', sinatra: true do
+    with_page(baseURL: "#{server_prefix}/url-construction/") do |page|
+      expect(page.goto('mypage.html').url).to eq("#{server_prefix}/url-construction/mypage.html")
+      expect(page.goto('./mypage.html').url).to eq("#{server_prefix}/url-construction/mypage.html")
+      expect(page.goto('/mypage.html').url).to eq("#{server_prefix}/mypage.html")
+      expect(page.goto('.').url).to eq("#{server_prefix}/url-construction/")
+      expect(page.goto('/').url).to eq("#{server_prefix}/")
+    end
+  end
+
+  it 'should not construct a new URL when valid URLs are passed', sinatra: true do
+    with_page(baseURL: 'https://example.com') do |page|
+      expect(page.goto(server_empty_page).url).to eq(server_empty_page)
+
+      page.goto('data:text/html,Hello world')
+      expect(page.evaluate('window.location.href')).to eq('data:text/html,Hello world')
```

Comment:
Original implementation uses `() => window.location.href`, however it returns nil.
In 1.12.0, it returns data:text/html,Hello world.

Reply:
It seems to be caused by https://github.com/microsoft/playwright/pull/7535

Reply:
1625813701000 doesn't include the bugfix.
1625872907000 includes the fix.